packages/vscode-extension/src/utils/commonUtils.ts

```diff
@@ -207,3 +207,7 @@ export function getAllFeatureFlags(): string[] | undefined {
 
   return result;
 }
+
+export function getIsUpgradeUser(): string | undefined {
+  return ext.context.globalState.get<string>(UserState.IsUpgrade);
```

Comment:
Why not check this.context.globalState.get<string>(SyncedState.Version) directly?

Reply:
both new users and upgraded users, their synced version will be update to latest version finally. so there is not way to distinguish is new users or upgrade users.

Reply:
1. Actually we can know that when we check the version change like you do now. But we can save it in local memory or as global state and you choose the latter one, right?
2. Another thing is if current user is upgrade user and the global state is set to true. Then next time when he start VSC and no version change, the global state is still true which will cause wrong telemtry. Let me know if I'm wrong

Reply:
to point 2: it is expected. if user already upgrade before, then it will always true. Maybe I need to change the **upgrade user** to **existing users**, which may remove the confusion.

Reply:
I got your point.